mongodb/activity/queryDocument/activity.go

```diff
@@ -0,0 +1,209 @@
+package queryDocument
+
+import (
+	ctx "context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	mongocon "github.com/project-flogo/datastore-contrib/mongodb/connector/connection"
+	// "github.com/TIBCOSoftware/flogo-lib/core/activity"
+	// "github.com/TIBCOSoftware/flogo-lib/core/data"
+	// "github.com/TIBCOSoftware/flogo-lib/logger"
+	//"github.com/TIBCOSoftware/flogo-lib/core/data"
+	//"github.com/TIBCOSoftware/flogo-lib/core/data"
+	// Need to remove this after testing with latest API
+	"github.com/project-flogo/core/activity"
+	"github.com/project-flogo/core/support/log"
+
+	//"github.com/project-flogo/core/data/metadata"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// var log = logger.GetLogger("activity-mongodb")
+
+const (
+	connectionProp = "mongoConnection"
+	ivOperation    = "operation"
+	outputProperty = "Output"
+	ivInput        = "input"
+	ovCount        = "count"
+)
+
+func init() {
+	fmt.Println("Entered the init")
+	err := activity.Register(&Activity{}, New)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+
+// New functioncommon
+func New(ctx1 activity.InitContext) (activity.Activity, error) {
+	fmt.Println("Entered the new method ")
+	settings := &Settings{}
+	//fmt.Println(ctx1.Settings())
+	//fmt.Println("Before Getting the config")
+	settings.Connection, _ = mongocon.GetSharedConfiguration(ctx1.Settings()["mongoConnection"])
+	//fmt.Println(ctx1.Settings()["mongoConnection"])
+	mcon, _ := settings.Connection.(*mongocon.MongodbSharedConfigManager)
+	//	mCtx, _ := ctx.WithTimeout(ctx.Background(), 10*time.Second)
+	client := mcon.GetClient()
+	// opts := options.Client()
+
+	// // if s.Username != "" && s.Password != "" {
+	// // 	opts = opts.SetAuth(options.Credential{
+	// // 		Username: s.Username,
+	// // 		Password: s.Password,
+	// // 	})
+	// // }
+	// //fmt.Println("Before connect")
+	// //fmt.Println(mcon.GetClientConfiguration().ConnectionURI)
+	// client, err := mongo.Connect(mCtx, opts.ApplyURI(mcon.GetClientConfiguration().ConnectionURI))
+
+	// if err != nil {
+	// 	fmt.Println("Error creating client")
+	// 	return nil, err
+	// }
+	//fmt.Println(client)
+	//db := client.Database(s.Database)
+	//collection := db.Collection(s.Collection)
+
+	act := &Activity{client: client, operation: ctx1.Settings()["operation"].(string), collectionName: ctx1.Settings()["collName"].(string), database: mcon.GetClientConfiguration().Database}
+	return act, nil
+}
+
+// Activity is a stub for your Activity implementation
+type Activity struct {
+	client         *mongo.Client
+	operation      string
```

Comment:
Can you rename `operation` to `method ` ?

Reply:
No, it's actually an operation of query. Users can choose one of the following three operations:
- Fine One Document
- Find Many Documents
Hence want to retain it as operation